cpp/src/arrow/flight/sql/odbc/entry_points.cc

```diff
@@ -109,3 +111,204 @@ SQLRETURN SQL_API SQLConnectW(SQLHDBC conn, SQLWCHAR* dsnName, SQLSMALLINT dsnNa
+                             SQLLEN bufferLength, SQLLEN* strLen_or_IndPtr) {
+  LOG_DEBUG(
+      "SQLBindCol called with statementHandle: {}, columnNumber: {}, targetType: {}, "
+      "targetValuePtr: {}, bufferLength: {}, strLen_or_IndPtr: {}",
+      statementHandle, columnNumber, targetType, targetValuePtr, bufferLength,
+      fmt::ptr(strLen_or_IndPtr));
+  return SQL_ERROR;
+}
+
+SQLRETURN SQL_API SQLCancel(SQLHSTMT statementHandle) {
+  LOG_DEBUG("SQLCancel called with statementHandle: {}", statementHandle);
+  return SQL_ERROR;
+}
+
+SQLRETURN SQL_API SQLCloseCursor(SQLHSTMT statementHandle) {
+  LOG_DEBUG("SQLCloseCursor called with statementHandle: {}", statementHandle);
+  return SQL_ERROR;
+}
+
+SQLRETURN SQL_API SQLColAttributeW(SQLHSTMT statementHandle, SQLUSMALLINT columnNumber,
+                                   SQLUSMALLINT fieldIdentifier,
+                                   SQLPOINTER characterAttributePtr,
+                                   SQLSMALLINT bufferLength, SQLSMALLINT* stringLengthPtr,
+                                   SQLLEN* numericAttributePtr) {
+  LOG_DEBUG(
+      "SQLColAttributeW called with statementHandle: {}, columnNumber: {}, "
+      "fieldIdentifier: {}, characterAttributePtr: {}, bufferLength: {}, "
+      "stringLengthPtr: {}, numericAttributePtr: {}",
+      statementHandle, columnNumber, fieldIdentifier, characterAttributePtr, bufferLength,
+      fmt::ptr(stringLengthPtr), fmt::ptr(numericAttributePtr));
+  return SQL_ERROR;
+}
+
+SQLRETURN SQL_API SQLColumnsW(SQLHSTMT statementHandle, SQLWCHAR* catalogName,
+                              SQLSMALLINT nameLength1, SQLWCHAR* schemaName,
+                              SQLSMALLINT nameLength2, SQLWCHAR* tableName,
+                              SQLSMALLINT nameLength3, SQLWCHAR* columnName,
+                              SQLSMALLINT nameLength4) {
+  LOG_DEBUG(
+      "SQLColumnsW called with statementHandle: {}, catalogName: {}, nameLength1: {}, "
+      "schemaName: {}, nameLength2: {}, tableName: {}, nameLength3: {}, columnName: {}, "
+      "nameLength4: {}",
+      statementHandle, fmt::ptr(catalogName), nameLength1, fmt::ptr(schemaName),
+      nameLength2, fmt::ptr(tableName), nameLength3, fmt::ptr(columnName), nameLength4);
+  return SQL_ERROR;
+}
+
+SQLRETURN SQL_API SQLError(SQLHENV handleType, SQLHDBC handle, SQLHSTMT hstmt,
+                           SQLWCHAR FAR* szSqlState, SQLINTEGER FAR* pfNativeError,
+                           SQLWCHAR FAR* szErrorMsg, SQLSMALLINT cbErrorMsgMax,
+                           SQLSMALLINT FAR* pcbErrorMsg) {
+  LOG_DEBUG(
+      "SQLError called with handleType: {}, handle: {}, hstmt: {}, szSqlState: {}, "
+      "pfNativeError: {}, szErrorMsg: {}, cbErrorMsgMax: {}, pcbErrorMsg: {}",
+      handleType, handle, hstmt, fmt::ptr(szSqlState), fmt::ptr(pfNativeError),
+      fmt::ptr(szErrorMsg), cbErrorMsgMax, fmt::ptr(pcbErrorMsg));
+  return SQL_ERROR;
+}
```

Comment:
```suggestion
SQLRETURN SQL_API SQLErrorW(SQLHENV handleType, SQLHDBC handle, SQLHSTMT hstmt,
                           SQLWCHAR FAR* szSqlState, SQLINTEGER FAR* pfNativeError,
                           SQLWCHAR FAR* szErrorMsg, SQLSMALLINT cbErrorMsgMax,
                           SQLSMALLINT FAR* pcbErrorMsg) {
  LOG_DEBUG(
      "SQLErrorW called with handleType: {}, handle: {}, hstmt: {}, szSqlState: {}, "
      "pfNativeError: {}, szErrorMsg: {}, cbErrorMsgMax: {}, pcbErrorMsg: {}",
      handleType, handle, hstmt, fmt::ptr(szSqlState), fmt::ptr(pfNativeError),
      fmt::ptr(szErrorMsg), cbErrorMsgMax, fmt::ptr(pcbErrorMsg));
  return SQL_ERROR;
}
```
Just realized that we should use SQLErrorW instead in here.